### PR TITLE
Added many missing properties to CommonTypes

### PR DESCRIPTION
**Pull request by livarcocc**
Added many missing properties to CommonTypes as well as attributes to PackageReference and a new item for FrameworkReference.

Fixes https://github.com/microsoft/msbuild/issues/3979

cc @DamianEdwards

### CODE REVIEW

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 1]
```diff
@@ -216,6 +216,72 @@ elementFormDefault="qualified">
...
+                <xs:annotation>
+                  <xs:documentation>
+                    <!-- _locID_text="FrameworkReference_IsTrimmable" _locComment="" -->Controls whether the runtime assets of this shared framework can be trimmed by the IL Linker (if PublishTrimmed is true).
+                  </xs:documentation>
+                </xs:annotation>
+              </xs:element>
+              <xs:element name="TargetLatestRuntimePatch">
+                <xs:annotation>
+                  <xs:documentation>
+                    <!-- _locID_text="FrameworkReference_TargetLatestRuntimePatch" _locComment="" -->Controls whether the app will target the latest patch of the runtime.  Defaults to true fro self-contained apps, false otherwise.
```
Comment (DamianEdwards) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Typo: `fro`

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 2]
```diff
@@ -216,6 +216,72 @@ elementFormDefault="qualified">
...
+            <xs:annotation>
+              <xs:documentation>
+                <!-- _locID_text="FrameworkReference_Attribute_IsTrimmable" _locComment="" -->Controls whether the runtime assets of this shared framework can be trimmed by the IL Linker (if PublishTrimmed is true).
+              </xs:documentation>
+            </xs:annotation>
+          </xs:attribute>
+          <xs:attribute name="TargetLatestRuntimePatch" type="xs:string" use="optional">
+            <xs:annotation>
+              <xs:documentation>
+                <!-- _locID_text="FrameworkReference_Attribute_TargetLatestRuntimePatch" _locComment="" -->Controls whether the app will target the latest patch of the runtime.  Defaults to true fro self-contained apps, false otherwise.
```
Comment (DamianEdwards) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Typo: `Defaults to true fro self-contained apps`

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 3]
```diff
@@ -1845,6 +1995,54 @@ elementFormDefault="qualified">
...
       </xs:annotation>
     </xs:element>
+    <xs:element name="RestoreSources" type="msb:StringPropertyType" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="RestoreSources" _locComment="" -->The feeds that NuGet will use for restoring packages for this project.</xs:documentation>
+      </xs:annotation>
+    </xs:element>
+    <xs:element name="PackageLicenseExpression" type="msb:StringPropertyType" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="PackageLicenseExpression" _locComment="" -->The project license's SPDX identifier. Only OSI and FSF approved licenses can use an identifier. Other licenses should use PackageLicenseFile. Read more about license metadata.</xs:documentation>
```
Comment (DamianEdwards) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Probably needs to lose the last sentence: "read more for..."

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 4]
```diff
@@ -216,6 +216,72 @@ elementFormDefault="qualified">
               </xs:documentation>
             </xs:annotation>
           </xs:attribute>
+          <xs:attribute name="IncludeAssets" type="xs:string" use="optional">
+            <xs:annotation>
+              <xs:documentation>
+                <!-- _locID_text="PackageReference_Attribute_IncludeAssets" _locComment="" -->Assets to include from this reference
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Do these match the NuGet docs for these? Should we link to those docs in the loc comment to ease translation consistency?

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 5]
```diff
@@ -1845,6 +1995,54 @@ elementFormDefault="qualified">
...
+      </xs:annotation>
+    </xs:element>
+    <xs:element name="TrimmerRootAssembly" type="msb:StringPropertyType" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="TrimmerRootAssembly" _locComment="" -->Indicates to the linker to explicitly keep an assembly by adding it to your csproj (use the assembly name without extension).</xs:documentation>
+      </xs:annotation>
+    </xs:element>
+    <xs:element name="TrimmerRootDescriptor" type="msb:StringPropertyType" substitutionGroup="msb:Property">
+      <xs:annotation>
+        <xs:documentation><!-- _locID_text="TrimmerRootDescriptor" _locComment="" -->Gives the linker a more specific list of types/methods, etc. to include using an xml file.</xs:documentation>
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]:
```suggestion
        <xs:documentation><!-- _locID_text="TrimmerRootDescriptor" _locComment="" -->Gives the linker a more specific list of types/methods, etc. to include. Path to an xml file.</xs:documentation>
```

?

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd [hunk 6]
```diff
@@ -216,6 +216,72 @@ elementFormDefault="qualified">
...
+            </xs:annotation>
+          </xs:attribute>
+        </xs:extension>
+      </xs:complexContent>
+    </xs:complexType>
+  </xs:element>
+  <xs:element name="FrameworkReference" substitutionGroup="msb:Item">
+    <xs:annotation>
+      <xs:documentation>
+        <!-- _locID_text="FrameworkReference" _locComment="" -->Reference to a shared framework.
```
Comment (rainersigwald) [inline src/MSBuild/Microsoft.Build.CommonTypes.xsd]: For consistency, drop trailing periods on all of these new ones.

**Comment by livarcocc** [comment]
Will work on changes suggest by @rainersigwald

File: src/MSBuild/Microsoft.Build.CommonTypes.xsd (hunk 6 above)
Comment (livarcocc) [reply src/MSBuild/Microsoft.Build.CommonTypes.xsd]: Well, this is super inconsistent already. There are a bunch that have periods and a bunch without. If we are going to make this consistent, I would rather make it consistent across the board. Probably by adding periods in the places without it.

@rainersigwald what do you think?

**Pull request merged**